packages/bemlinter/src/lint.js

```diff
@@ -125,4 +130,13 @@ module.exports = (sources, userOptions = defaultOptions) => {
       }
     });
   }
+  
+  function checkSelector($, filePath, blockName) {
+    eachWrapper($('selector'), wrapper => {
+      const selector = nodeToString(wrapper.node).trim();
+      if (/&[-_a-z]/.test(selector)) {
```

Comment:
Also from a top level point of view, I'd rather use an exclusive pattern or deal with it find & occurrences and checking next char or object.